app/models/customer.rb

```diff
@@ -1,5 +1,20 @@
 class Customer < ApplicationRecord
 
   has_many :invoices
+  has_many :transactions, through: :invoices
 
-end 
+  def self.top_customers
+    joins(:transactions)
+    .where('result = ?', 1)
+    .group(:id)
+    .select("customers.*, count('transactions.result') as top_result")
+    .order(top_result: :desc)
+    .limit(5)
+  end
+
+  def number_of_transactions
```

Comment:
This method is redundant. Since you've already counted the transactions in the top_customers you can access it in your view by just calling c.top_result